utils/devel/extract-perl

```diff
@@ -1,132 +1,123 @@
 #!/usr/bin/perl
 
-=pod
+=description
 
-This script contains a relatively simple state machine to scan various
-Perl files for translatable strings.
+This script scans various Perl files for translatable strings.
 
-The scanner consists of 2 main state processes (block comment (POD) and
-text() call parsing).
-
-Single line comments are intentionally *not* stripped, because single line
-comments sometimes intentionally contain translatable strings.
+The scanner uses the Perl Pattern Recognizer to find text() or marktext()
+translation routines, either in code or comments, because they sometimes
+intentionally contain translatable strings.
 
 This script exists because xgettext and xgettext.pl don't allow us to
 extract a sub-set of strings from our SQL files.
 
 =cut
 
-
 use strict;
 use warnings;
 
+use PPR;
+use utf8;
+use Carp qw( croak );
+use Getopt::Long;
 
-use constant {
-    NUL => 1,
-    POD => 10,
-    TXT => 20,
-    TXT_FIN => 21,
-};
-
+#TODO: What relevant options to add?
+my $pack;
+  GetOptions ("pack"  => \$pack)     # pack references
```

Comment:
--pack isn't a supported format by poEdit and other po editors. No need to make things more complex by including it (right?).

Reply:
Pack shrinks LedgerSMB.pot to 62% of actual size.